File: library/src/main/java/me/shkschneider/skeleton/ui/Toaster.java
package me.shkschneider.skeleton.ui;

import android.support.annotation.NonNull;
import android.text.TextUtils;
import android.widget.Toast;

import me.shkschneider.skeleton.helper.ContextHelper;
import me.shkschneider.skeleton.helper.LogHelper;
import me.shkschneider.skeleton.helper.ThreadHelper;

public class Toaster {

    public static void lengthShort(@NonNull final String msg) {
        toast(msg, Toast.LENGTH_SHORT);
    }

    public static void lengthLong(@NonNull final String msg) {
        toast(msg, Toast.LENGTH_LONG);
    }

    private static void toast(@NonNull final String msg, final int duration) {
        if (TextUtils.isEmpty(msg)) {
            LogHelper.warning("Message was NULL");
        }
        ThreadHelper.foregroundThread(new Runnable() {
            @Override
            public void run() {
                Toast.makeText(ContextHelper.applicationContext(), msg, duration).show();
            }
        });
    }

}

File: library/src/main/kotlin/me/shkschneider/skeleton/ui/Toaster.kt
package me.shkschneider.skeleton.ui

import android.support.annotation.UiThread
import android.widget.Toast

import me.shkschneider.skeleton.helper.ContextHelper
import me.shkschneider.skeleton.helper.Logger

object Toaster {

    @UiThread
    fun show(msg: String, duration: Int = Toast.LENGTH_LONG) {
        if (msg.isBlank()) {
            Logger.warning("Message was NULL")
        }
        Toast.makeText(ContextHelper.applicationContext(), msg, duration).show()
    }

}
